lib/node/disseminator.mli

```diff
@@ -0,0 +1,37 @@
+(** Component responsible for gossip-style dissemination of
+    messages across the network *)
+
+(** A record containing information and state relevant to the
+    dissemination component. *)
+type t
+
+(** Creates a dissemination component that can be attached to a node
+    when given the number of "rounds" for which each new message should
+    be disseminated and an "epoch length", in seconds, which determines
+    whether a message is too old to be disseminated again by checking
+    whether the message is newer than n seconds old, where n is the
+    given epoch length. *)
+val create : int -> float -> t
```

Comment:
What about creating custom types to make this signature more readable?
Like:
```ocaml
type rounds = int
type epoch_length = float

val create : rounds -> epoch_length -> t
```
I think we have several functions where such custom types would help the readability, what do you think?

Reply:
We can also just use named parameters, the effect is the same